DOCSP-49621 Customize cluster settings (#522)
mongodb/docs-golang: source/connect/connection-options/cluster-settings.txt

.. _golang-cluster-settings:

==========================

> Customize Cluster Settings

Author: Since Go doesn't support the server selection method, I modeled this page after the Java page which has similar functionality (linked in the ticket) which uses this title. We might need to update the Unified ToC spreadsheet to reflect this alternate option though.

           load balancer.

           *Default:* ``false``

       * - ``SetSRVServiceName()``
         - Specifies a custom service name of the `SRV resource records
           <https://www.rfc-editor.org/rfc/rfc2782>`__ the driver retrieves to
           construct your :manual:`seed list
           </reference/glossary/#std-term-seed-list>`. To use a custom SRV
           service name in SRV discovery, you must call this function before you
           call ``ApplyURI()``.

    To learn more about the available methods, see the :ref:`golang-cluster-settings-resources` section.

Example
~~~~~~~

Select the :guilabel:`Connection String` or :guilabel:`ClientOptions`  tab to
see the corresponding example:

.. tabs::

    .. tab:: Connection String
      :tabid: uriExample

      The following code uses the connection string to configure the maximum
      server selection timeout to 10 seconds and the local threshold to 20
      milliseconds:

      .. literalinclude:: /includes/connect/cluster-settings-uri.go
        :language: go
        :start-after: start-uri-variable
        :end-before: end-uri-variable
        :dedent:

>       The following code creates a client and passes the connection string to the

Author: I opted to break these up in order to use the approach of setting a constant variable for the URI that a previous tech reviewer suggested, while avoiding having to show the entire main() funciton.

>       The following code creates a client and passes the connection string to the

Reviewer: makes sense!

.. _golang-cluster-settings:

==========================
Customize Cluster Settings
==========================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::

>    :keywords: code example, read preference, write

Reviewer: S: consider making the keywords more specific to clusters/connections, since I don't think read preferences are mentioned

Overview
--------

In this guide, you can learn how the {+driver-short+} manages clusters and how
to customize the cluster settings.

Specify Settings
----------------

You can specify settings for your clusters by using either a connection string
or a ``ClientOptions`` struct when creating a new ``Client``
instance. Select the :guilabel:`Connection String` or :guilabel:`ClientOptions` tab to see the
available options.

.. tabs::

  .. tab:: Connection String
    :tabid: uri

    The following table describes the parameters you can use in your connection
    string to modify the driver's behavior when interacting with your MongoDB
    cluster:

    .. list-table::
       :header-rows: 1
       :stub-columns: 1
       :widths: 40 60

>        * - Method

Reviewer: S: I'd use "parameter" or "option"
```suggestion
       * - Parameter
```

           *Default:* 30 seconds

       * - ``localThresholdMS``
         - Specifies the maximum latency in milliseconds for selecting a server.
           
           *Default:* 15 milliseconds

       * - ``replicaSet``
         - Specifies the name of the replica set to connect to.

       * - ``directConnection``
         - Specifies whether to connect directly to a single server, bypassing the
           replica set or sharded cluster.

           *Default:* ``false``

       * - ``loadBalanced``
         - Specifies whether the driver is connecting to MongoDB using a
           load balancer.

           *Default:* ``false``

       * - ``srvServiceName``
         - Specifies the service name of the `SRV resource records
           <https://www.rfc-editor.org/rfc/rfc2782>`__ the driver retrieves to
           construct your :manual:`seed list
           </reference/glossary/#std-term-seed-list>`. You must use the
           :manual:`DNS Seed List Connection Format
           </reference/connection-string/#srv-connection-format>` in
           your connection string to use this option.

  .. tab:: ClientOptions
    :tabid: options

>     The following table describes several methods you can chain to your settings to

Reviewer: S: might be helpful to clarify what you mean by "settings"
```suggestion
    The following table describes several methods you can chain to your ``ClientOptions`` struct to
```

    modify the driver's behavior:

    .. list-table::
       :header-rows: 1
       :stub-columns: 1
       :widths: 40 60

>        * - Field

Reviewer: S:
```suggestion
       * - Method
```
